samples/Azure.ResourceManager.Sample/Generated/RestOperations/AvailabilitySetsRestOperations.cs

```diff
@@ -105,7 +105,7 @@ public async Task<Response<AvailabilitySetData>> CreateOrUpdateAsync(string subs
                         return Response.FromValue(value, message.Response);
                     }
                 default:
-                    throw await ClientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
+                    throw new RequestFailedException(message.Response);
```

Comment:
Can we remove the ClientDiagnostics property / ctor parameter since its not needed any more

Reply:
We need ClientDiagnostics property and ctor parameter when we generate protocol method in RestClient for some cases for example: [ProtocolMethodsInRestClient/Generated/TestServiceRestClient.cs](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ProtocolMethodsInRestClient/Generated/TestServiceRestClient.cs#L136)

Reply:
is it possible to detect that and omit if there are none of those?  I believe its probably the majority case that it isn't needed